components/finsh/msh.h

```diff
@@ -19,4 +19,140 @@ void msh_auto_complete(char *prefix);
+                RT_USED const struct msh_syscall __fsym_##cmd RT_SECTION("FSymTab")= \
+                {                           \
+                    __fsym_##cmd##_name,    \
+                    __fsym_##cmd##_desc,    \
+                    (syscall_func)&name     \
+                };
+
+#endif
+#else
+#ifdef _MSC_VER
+#define MSH_FUNCTION_EXPORT_CMD(name, cmd, desc)      \
+                const char __fsym_##cmd##_name[] = #cmd;            \
+                __declspec(allocate("FSymTab$f"))                   \
+                const struct msh_syscall __fsym_##cmd =           \
+                {                           \
+                    __fsym_##cmd##_name,    \
+                    (syscall_func)&name     \
+                };
+#pragma comment(linker, "/merge:FSymTab=mytext")
+
+#elif defined(__TI_COMPILER_VERSION__)
+#define MSH_FUNCTION_EXPORT_CMD(name, cmd, desc)      \
+                __TI_MSH_EXPORT_FUNCTION(__fsym_##cmd);           \
+                const char __fsym_##cmd##_name[] = #cmd;            \
+                const struct msh_syscall __fsym_##cmd =           \
+                {                           \
+                    __fsym_##cmd##_name,    \
+                    (syscall_func)&name     \
+                };
+
+#else
+#define MSH_FUNCTION_EXPORT_CMD(name, cmd, desc)                      \
+                const char __fsym_##cmd##_name[] = #cmd;                            \
+                RT_USED const struct msh_syscall __fsym_##cmd RT_SECTION("FSymTab")= \
+                {                                                                   \
+                    __fsym_##cmd##_name,                                            \
+                    (syscall_func)&name                                             \
+                };
+
+#endif
+#endif /* end of MSH_USING_DESCRIPTION */
+#endif /* end of MSH_USING_SYMTAB */
+
+/**
+ * @ingroup msh
+ *
+ * This macro exports a command to module shell.
+ *
+ * @param command is the name of the command.
+ * @param desc is the description of the command, which will show in help list.
+ */
+#define MSH_CMD_EXPORT(command, desc)   \
+    MSH_FUNCTION_EXPORT_CMD(command, command, desc)
+
+/**
+ * @ingroup msh
+ *
+ * This macro exports a command with alias to module shell.
+ *
+ * @param command is the name of the command.
+ * @param alias is the alias of the command.
+ * @param desc is the description of the command, which will show in help list.
+ */
+#define MSH_CMD_EXPORT_ALIAS(command, alias, desc)  \
+    MSH_FUNCTION_EXPORT_CMD(command, alias, desc)
+
+/* system call table */
+struct msh_syscall
+{
+    const char     *name;       /* the name of system call */
+#if defined(MSH_USING_DESCRIPTION) && defined(MSH_USING_SYMTAB)
+    const char     *desc;       /* description of system call */
+#endif
+    syscall_func func;      /* the function address of system call */
+};
```

Comment:
Please use `msh_cmd` to replace `msh_syscall`; Please use `cmd_func` to replace `syscall_func`.